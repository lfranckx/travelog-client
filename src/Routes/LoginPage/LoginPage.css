.login-section {
    margin: 0 auto;
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    max-width: 400px;
}


.login-form .name-input {
    justify-content: center;
    
}

.login-form label, .login-form input {
    font-size: 20px;
}

.login-section > div {
    text-align: left;
}