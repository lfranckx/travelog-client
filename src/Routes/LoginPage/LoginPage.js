import React, { Component } from 'react';
import LoginForm from '../../Components/LoginForm/LoginForm';
import './LoginPage.css';

export default class LoginPage extends Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        }
    }

    handleSubmitSuccess = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || '/';
        history.push(destination);
    }

    render() {
        return (
            <section className="login-section">
                <h2>Login</h2>
                <LoginForm 
                    onSubmitSuccess={this.handleSubmitSuccess}
                />
            </section>
        );
    }
}