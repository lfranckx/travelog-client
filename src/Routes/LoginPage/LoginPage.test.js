import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from './LoginPage';

describe(`LoginPage component`, () => {

    it(`renders log in page without crashing`, () => {
        const wrapper = shallow(<LoginPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})