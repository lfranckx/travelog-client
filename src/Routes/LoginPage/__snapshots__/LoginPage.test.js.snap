// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders log in page without crashing 1`] = `
<section
  className="login-section"
>
  <h2>
    Login
  </h2>
  <LoginForm
    onSubmitSuccess={[Function]}
  />
</section>
`;
